designs/Verify-Verbosity.md

```diff
@@ -0,0 +1,416 @@
+```
+
+</details>
+
+<details>
+<summary>After incident output</summary>
+
+```
+dotnet nuget verify "package.nupkg" -v n
+Verifying packageA.1.0.0
+C:\Users\kapenaga\Downloads\package.nupkg
+Signature Hash Algorithm: SHA256
+
+Signature type: Author
+Verifying the author primary signature with signing certificate:
+  Subject name: CN=test
+  SHA1 hash: B0A2B3B1695AB8361B1D2B14A9F5D64136E26380
+  SHA256 hash: 89A2B6EB529E0AEBF0D11C8A18A846C7B8D1290791B6BF494BAFEC299F2EAAB2
+  Issued by: CN=test
+  Valid from: 1/29/2021 12:28:11 PM to 02/29/2021 1:28:11 PM
+Timestamp: 2/3/2021 3:36:12 PM
+Verifying author primary signature's timestamp with timestamping service certificate:
+  Subject name: CN=NuGet Test Root Certificate Authority (40998d55-3d73-4a3b-a689-55e30c1fac3c), O=NuGet, L=Redmond, S=WA, C=US
+  SHA1 hash: 6B2378A3DC9CA185252BB66F24F262D129165B5B
+  SHA256 hash: 61B18DE3D814FA7960C6ED62DB20BEA6D0F8D65F678464D7D7C9227E7D5DEFBD
+  Issued by: CN=NuGet Test Root Certificate Authority (40998d55-3d73-4a3b-a689-55e30c1fac3c), O=NuGet, L=Redmond, S=WA, C=US
+  Valid from: 2/3/2021 3:36:11 PM to 12/31/2099 4:00:00 PM
+
+error: NU3028: The author primary signature's timestamp signature's certificate is not trusted by the trust provider.
+error: Package signature validation failed.
+
+Finished with 1 errors and 0 warnings.
+```
+
+</details>
+
+#### `Debian case - Verifying expired author signed package with untrusted timestamping signing certificate`
+
+<details>
+<summary>Before incident output</summary>
+
+```
+dotnet nuget verify "AuthorExpired.1.0.0.nupkg" -v n
+Verifying AuthorExpired.1.0.0
+C:\Users\kapenaga\Downloads\signed-packages\AuthorExpired.1.0.0.nupkg
+Signature Hash Algorithm: SHA256
+Timestamp: 2/3/2021 3:36:12 PM
+Verifying author primary signature's timestamp with timestamping service certificate:
+  Subject name: CN=NuGet Test Root Certificate Authority (40998d55-3d73-4a3b-a689-55e30c1fac3c), O=NuGet, L=Redmond, S=WA, C=US
+  SHA1 hash: 6B2378A3DC9CA185252BB66F24F262D129165B5B
+  SHA256 hash: 61B18DE3D814FA7960C6ED62DB20BEA6D0F8D65F678464D7D7C9227E7D5DEFBD
+  Issued by: CN=NuGet Test Root Certificate Authority (40998d55-3d73-4a3b-a689-55e30c1fac3c), O=NuGet, L=Redmond, S=WA, C=US
+  Valid from: 2/3/2021 3:36:11 PM to 12/31/2099 4:00:00 PM
+error: NU3028: The author primary signature's timestamp found a chain building issue: UntrustedRoot: A certificate chain processed, but terminated in a root certificate which is not trusted by the trust provider.
+Finished with 1 errors and 0 warnings.
+error:
+error: Package signature validation failed.
+```
+
+[Damon Tivel](https://github.com/dtivel) created an issue <https://github.com/NuGet/Home/issues/10535>
+
+</details>
+
+<details>
+<summary>After incident output</summary>
+
+```
+dotnet nuget verify "AuthorExpired.1.0.0.nupkg" -v n
+Verifying packageA.1.0.0
+C:\Users\kapenaga\Downloads\package.nupkg
+Signature Hash Algorithm: SHA256
+
+Signature type: Author
```

Comment:
Anyway to make this more human-readable? Some thoughts:

- Bold where needed
- Use color with intention
- ASCII art where it makes sense
- Symbols or emoji if they apply

Reply:
Main reason is that I think many people interpret this output as already very verbose even with minimal output. Bolding, spacing, and coloring the important parts I think would help with verbosities.

Reply:
@JonDouglas - Thanks for the feedback, I really like your thoughts on how to make the log messages human-readable. TBH, I think we should definitely consider these suggestions for all the `dotnet` commands in another effort because `Logging framework is shared` and any change impacts other commands. We should bring up these ideas with `dotnet` team as they own the CLI experience. Please let me know what do you think, I will be happy to engage in further conversations.

Reply:
Absolutely. This is a wonderful idea.